registry/storage/manifeststore.go

```diff
@@ -13,6 +15,149 @@ import (
 	"github.com/docker/distribution/manifest/schema2"
 )
 
+const (
+	// MediaTypeOrphanedManifestRevision specifies the mediaType for orphaned
+	// (unlinked, delted) manifest that can be cleaned up. This type of
+	// manifest can be returned only during enumeration if enabled with option.
+	MediaTypeOrphanedManifestRevision = "application/vnd.docker.distribution.manifest.orphaned.v2+json"
```

Comment:
This is a bit of a hack. We need to somehow represent broken manifest revisions and be able to enumerate them. For the purpose of enumeration there's a [manifest service option ](https://github.com/miminar/distribution/commit/12c3fac4f69620714775e10053eeb734f2a2339c#diff-e566b0debd4b786531022c5420d4220aR174).

To represent broken manifest revision, `Enumerate()` method constructs a `schema2.DeserializedManifest` with this mediatype and a revision digest set.

This is clearly an abuse of the schema, but we know that such a broken manifest can be received only by a caller expecting it.

Other posibilities to represent this scenario that come to my mind are:

1. create a new manifest schema solely for this purpose - this is IMHO too heavyweight and since it needs to be exported, it would clutter the `manifest` namespace with something clients could not use
2. revert back to digest enumeration e.g.:
  - `Enumerate(ctx context.Context) ([] digest.Digest err error)`
  - `Enumerate(ctx context.Context, ingester func(digest.Digest) error) error`